src/main/java/seedu/findvisor/model/person/NameContainsKeywordPredicate.java

```diff
@@ -0,0 +1,48 @@
+package seedu.findvisor.model.person;
+
+import java.util.function.Predicate;
+
+import seedu.findvisor.commons.util.StringUtil;
+import seedu.findvisor.commons.util.ToStringBuilder;
+
+/**
+ * A predicate for evaluating if a {@link Person}'s name contains (case-insensitive) a given keyword.
+ * This is used to filter for persons based on their name attribute.
+ */
+public class NameContainsKeywordPredicate implements Predicate<Person> {
```

Comment:
Maybe the class name should be changed to `NameStartsWithKeywordPredicate` so that it is more representative of the predicate?